tools/db/build/build_models_script.php

```diff
@@ -0,0 +1,330 @@
+        $this->generate_model_link_inserts() .
+        $this->generate_model_related_inserts();
+    }
+    
+    /**
+      Loads a .model_info file, parses it into the 
+      models array
+    */
+    function process_file($file) {
+      if(($data = file_get_contents($file)) === false) {
+        return false;
+      }
+      
+      $model = json_decode($data);
+      $model->json = $data;
+      
+      array_push($this->models, $model);     
+      return true;
+    }
+    
+    /**
+      Generate an SQL script to insert entries in to the t_model table
+    */
+    function generate_model_inserts() {
+      $result = <<<END
+        INSERT t_model (
+          model_id, 
+          name, 
+          author_name, 
+          author_email, 
+          description, 
+          license, 
+          last_modified, 
+          version,
+
+          min_keyman_version,
+          min_keyman_version_1,
+          min_keyman_version_2,
+          
+          package_filename,
+          package_filesize,
+          js_filename,
+          js_filesize,
+
+          is_rtl,
+          
+          includes_fonts,
+          
+          model_info
+        ) VALUES
+END;
+
+      $comma = '';
+      foreach($this->models as $model) {
+        $isRTL = isset($model->isRTL) && $model->isRTL;
+        
+        if(isset($model->minKeymanVersion) && preg_match('/^(\d+)\.(\d+)/', $model->minKeymanVersion, $matches)) {
+          $minKeymanVersion1 = $matches[1];
+          $minKeymanVersion2 = $matches[2];
+        } else {
+          $minKeymanVersion1 = 9;
+          $minKeymanVersion2 = 0;
+        }
+
+        $includesFonts = isset($model->packageIncludes) && in_array('fonts', $model->packageIncludes);
+        
+        $result .= <<<END
+$comma
+          ({$this->sqlv($model, 'id')},
+          {$this->sqlv($model, 'name')},
+          {$this->sqlv($model, 'authorName')},
+          {$this->sqlv($model, 'authorEmail')},
+          {$this->sqlv($model, 'description')},
+          {$this->sqlv($model, 'license')},
+          {$this->sqld($model, 'lastModifiedDate')},
+          {$this->sqlv($model, 'version')},
+          
+          {$this->sqlv($model, 'minKeymanVersion')},
+          $minKeymanVersion1,
+          $minKeymanVersion2,
+    
+          {$this->sqlv($model, 'packageFilename')},
+          {$this->sqli($model, 'packageFileSize')},
+          {$this->sqlv($model, 'jsFilename')},
+          {$this->sqli($model, 'jsFileSize')},
+
+          {$this->sqlb($isRTL)},
+    
+          {$this->sqlb($includesFonts)},
+
+          {$this->sqlv($model, 'json')})
+END;
+        $comma=',';
+      }
+
+      if($comma == '') return ''; // no entries found
+      return $result . ";\n";
+    }
+    
+    /**
+      Generate an SQL script to insert entries in to the t_model_language table
+    */
+    function generate_model_language_inserts() {
+      $result = <<<END
+        INSERT t_model_language (
+          model_id, 
```

Comment:
Is now a good time to set aside other fields like "display name" or the variants portion of the bcp47?

Reply:
no, let's keep that in separate PR work.